extensions-core/core/deployment/src/main/java/org/apache/camel/quarkus/core/deployment/CamelContextProcessor.java

```diff
@@ -165,17 +165,12 @@ public CamelRuntimeBuildItem runtime(
             List<CamelRuntimeTaskBuildItem> runtimeTasks,
             CamelConfig config) {
 
-        for (CamelRoutesBuilderClassBuildItem item : routesBuilderClasses) {
-            // don't add routes builders that are known by the container
-            if (containerBeans.getClasses().contains(item.getDotName())) {
-                continue;
-            }
-
-            recorder.addRoutes(context.getCamelContext(), item.getDotName().toString());
```

Comment:
I've spread the logic from the loop above in the code block below. So, the logic is now executed only when config.routesDiscovery.enabled = true.

However, to my current understanding, when config.routesDiscovery.enabled = false, then the routesBuilderClasses list is empty. As such, the logic in the loop above was not executed when config.routesDiscovery.enabled = false.